bmorph/evaluation/simple_river_network.py

```diff
@@ -189,7 +257,34 @@ def clear_network(self):
         """
         self.adj_mat = np.zeros(shape = (0))
         self.outlet.upstream = list()
+    
+    
+    def clear_end_markers(self,node):
+        """
+        clear_end_markers
+            sets all end_mark in nodes at and upstream of node
+            to False
+        """
+        if node:
+            node.end_marker = False
+            if node.upstream:
```

Comment:
Can remove this conditional too

Reply:
This I worry about since a non-existent node is still a NoneType object and will through an error since it is using recursion and NoneType does not have attribute end_marker

Reply:
*or if you meant line 270, then can do

Reply:
The `if node` is fine - but you should be able to remove the `if node.upstream:` since if `node` is `None` it will not go past the first conditional.